examples/source/e-commerce/Shopping_Cart/api.js

```diff
@@ -0,0 +1,201 @@
+    const queryString = 'id=' + id + '&name=' + encodeURIComponent(name) + '&price=' + price
+        + '&color=' + color + '&size=' + size + '&quantity=' + quantity + '&origin='
+        + encodeURIComponent(origin);
+    res.header('AMP-Redirect-To', req.protocol + '://' + req.get('host') +
+        `/documentation/examples/e-commerce/add_to_cart?${queryString}`);
+  } else {
+    // Complete the request from the origin
+    updateShoppingCartOnSession(req, id, name, price, color, size, quantity);
+    res.header('AMP-Redirect-To', req.protocol + '://' + req.get('host') +
+        '/documentation/examples/e-commerce/shopping_cart/');
+  }
+
+  enableCors(req, res);
+  res.json({});
+};
+
+function addToCartHandlerGet(req, res) {
+  const id = req.query.id;
+  const name = req.query.name;
+  const price = req.query.price;
+  const color = req.query.color;
+  const size = req.query.size;
+  const quantity = req.query.quantity;
+
+  updateShoppingCartOnSession(req, id, name, price, color, size, quantity);
+  res.redirect(req.protocol + '://' + req.get('host') + '/documentation/examples/e-commerce/shopping_cart/');
+};
+
+function cartItemsHandler(req, res) {
+  let shoppingCart = req.session.shoppingCart;
+
+  // cookie exists, but cart is empty
+  if (shoppingCart) {
+    shoppingCart = deserialize(shoppingCart);
+    console.log('Session exists.');
+  } else {
+    shoppingCart = createCart();
+    req.session.shoppingCart = serializer(shoppingCart);
+    console.log('Session doesn\'t exist.');
+  }
+
+  // wrap the shopping cart into an 'items' array, so it can be consumed with amp-list.
+  const shoppingCartResponse = {items: []};
+  shoppingCartResponse.items.push(shoppingCart);
+
+  enableCors(req, res);
+  res.send(shoppingCartResponse);
+};
+
+function deleteCartItemHandler(req, res) {
+  const id = req.fields.id;
+  const size = req.fields.size;
+
+  const shoppingCartResponse = {items: []};
+
+  let shoppingCart = req.session.shoppingCart;
+
+  if (shoppingCart) {
+    shoppingCart = deserialize(shoppingCart);
+    shoppingCart.removeItem(id, size);
+    req.session.shoppingCart = serializer(shoppingCart);
+    shoppingCartResponse.items.push(shoppingCart);
+  }
+
+  enableCors(req, res);
+  res.send(shoppingCartResponse);
+};
+
+function updateShoppingCartOnSession(req, id, name, price, color, size, quantity) {
+  const cartProduct = createCartItem(id, name, price, color, size, quantity);
+  let shoppingCart = req.session.shoppingCart;
+
+  if (shoppingCart) {
+    shoppingCart = deserialize(shoppingCart);
+  } else {
+    shoppingCart = createCart();
+  }
+
+  shoppingCart.addItem(cartProduct);
+  req.session.shoppingCart = serializer(shoppingCart);
+}
+
+
+function createCartItem(id, name, price, color, size, quantity) {
+  const cartProduct = {};
+  cartProduct.id = id;
+  cartProduct.name = name;
+  cartProduct.price = price;
+  cartProduct.color = color;
+  cartProduct.size = size;
```

Comment:
simpler:

```
const cartProduct = {
  id,
  name,
 ....
};
```

Reply:
Since we pass an object now, this is no longer needed.